src/Microsoft.Dnx.DesignTimeHost/Models/OutgoingMessages/DiagnosticMessageView.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Dnx.Runtime;
+
+namespace Microsoft.Dnx.DesignTimeHost.Models.OutgoingMessages
+{
+    public class DiagnosticMessageView
+    {
+        public DiagnosticMessageView(DiagnosticMessage data)
+        {
+            ErrorCode = data.ErrorCode;
+            SourceFilePath = data.SourceFilePath;
+            Message = data.Message;
+            Severity = data.Severity;
+            StartLine = data.StartLine;
+            StartColumn = data.StartColumn;
+            EndLine = data.EndLine;
+            EndColumn = data.EndColumn;
+            FormattedMessage = data.FormattedMessage;
+
+            var description = data.Source as LibraryDescription;
+            if (description != null)
+            {
+                Source = new
+                {
+                    Name = description.Identity.Name,
+                    Version = description.Identity.Version?.ToString()
+                };
+            }
+        }
+
+        public string ErrorCode { get; }
+
+        public string SourceFilePath { get; }
+
+        public string Message { get; }
+
+        public DiagnosticMessageSeverity Severity { get; }
+
+        public int StartLine { get; }
+
+        public int StartColumn { get; }
+
+        public int EndLine { get; }
+
+        public int EndColumn { get; }
+
+        public string FormattedMessage { get; }
+
+        public object Source { get; }
+
+        public override bool Equals(object obj)
+        {
+            var other = obj as DiagnosticMessageView;
+            
+            return other != null &&
+                   Severity == other.Severity &&
+                   StartLine == other.StartLine &&
+                   StartColumn == other.StartColumn &&
+                   EndLine == other.EndLine &&
+                   EndColumn == other.EndColumn &&
+                   string.Equals(ErrorCode, other.ErrorCode, StringComparison.Ordinal) &&
+                   string.Equals(SourceFilePath, other.SourceFilePath, StringComparison.Ordinal) &&
+                   string.Equals(Message, other.Message, StringComparison.Ordinal) &&
+                   string.Equals(FormattedMessage, other.FormattedMessage, StringComparison.Ordinal);
```

Comment:
Then it complicate the serialization. If the serialization is not treated specially, the `Name` and `Version` end up on the same level as other error property. 
```
    {
      "ErrorCode": "NU1002",
      "SourceFilePath": "C:\\Users\\trdai\\Documents\\Visual Studio 14\\Projects\\ConsoleApp1\\src\\ConsoleApp1\\project.json",
      "Message": "The dependency Newtonsoft.Json 4.5.11 in project ConsoleApp1 does not support framework DNXCore,Version=v5.0.",
      "Severity": 2,
      "StartLine": 10,
      "StartColumn": 24,
      "EndLine": 10,
      "EndColumn": 24,
      "FormattedMessage": "C:\\Users\\trdai\\Documents\\Visual Studio 14\\Projects\\ConsoleApp1\\src\\ConsoleApp1\\project.json(10,24): error NU1002: The dependency Newtonsoft.Json 4.5.11 in project ConsoleApp1 does not support framework DNXCore,Version=v5.0.",
      "Name": "Newtonsoft.Json",
      "Version": "4.5.11"
    }
```
It's misleading and also prevent from us to change `Source` type in the future. 

If we want to keep the two properties under `Source` then a simple `JToken.FromObject` doesn't cut.